Synapse3.SynapseModule/Patches/PlayerPatches/PlayerPatches.cs

```diff
@@ -453,10 +453,37 @@ public static bool OnUseWorkstation(WorkstationController __instance, ReferenceH
             return true;
         }
     }
+    [HarmonyPrefix]
+    [HarmonyPatch(typeof(CheckpointKiller), nameof(CheckpointKiller.OnTriggerEnter))]
+    public static bool OnCheckpointEnter(Collider other)
+    {
+        try
+        {
+            DecoratedPlayerPatches.OnFallingIntoAbyss(other);
+            return false;
+        }
+        catch (Exception ex)
+        {
+            NeuronLogger.For<Synapse>().Error("Sy3 Event: FallingIntoAbyss event failed\n" + ex);
+            return true;
+        }
+    }
 }
 
 internal static class DecoratedPlayerPatches
 {
+    public static void OnFallingIntoAbyss(Collider other)
+    {
+        var player = other.GetComponentInParent<ReferenceHub>();
```

Comment:
It's a small thing, but it would be a little more efficient to get the SynapsePlayer directly. The cast later just calls another GetComponent<> that isn't really needed in this case.

```suggestion
        var player = other.GetComponentInParent<SynapsePlayer>();
```